firebase-installations/src/androidTest/java/com/google/firebase/installations/FirebaseInstallationsInstrumentedTest.java

```diff
@@ -277,6 +279,36 @@ public void testGetId_PersistedFidError_BackendOk() throws InterruptedException
     }
   }
 
+  @Test
+  public void testGetId_expiredAuthToken_refreshesAuthToken() throws Exception {
+    // Update local storage with fid entry that has auth token expired.
+    persistedFid.insertOrUpdatePersistedFidEntry(EXPIRED_AUTH_TOKEN_ENTRY);
+    FirebaseInstallations firebaseInstallations =
+        new FirebaseInstallations(
+            mockClock, executor, firebaseApp, backendClientReturnsOk, persistedFid, mockUtils);
+
+    assertWithMessage("getId Task fails.")
+        .that(Tasks.await(firebaseInstallations.getId()))
+        .isNotEmpty();
+    PersistedFidEntry entryValue = persistedFid.readPersistedFidEntryValue();
+    assertWithMessage("Persisted Fid doesn't match")
+        .that(entryValue.getFirebaseInstallationId())
+        .isEqualTo(TEST_FID_1);
+
+    // Waiting for Task that generates auth token with the FIS Servers
+    executor.awaitTermination(500, TimeUnit.MILLISECONDS);
```

Comment:
really don't like this (feels like a weird hack when the test controls the underlying executorservice), but it's more of a theoretical concern than a pragmatic one

Reply:
One way to solve this could be allowing the getID implementation that takes a listener public. For this, we need Android SDK review( I not 100% sure). 

@ciarand  and @diwu-arete  wdyt?

Reply:
probably, probably not worth it